### ISSUE TITLE
GPU-Util varying considerably during training

### ISSUE THREAD

**Issue by lixiangchun** [AlexeyAB/darknet#465]
I trained `resnet50_yolo` with code clone from `AlexeyAB/darknet` and `pjreddie/darknet`, respectively. I found that GPU utility of `AlexeyAB/darknet` was varying considerably (e.g. oscillate between 50% - 98%), whereas GPU utility of `pjreddie/darknet` kept higher than 95%. `pjreddie/darknet` runs faster than `AlexeyAB/darknet`. This observation was also found when training `darknet19_yolo`.

**Comment by sivagnanamn** [comment]
Are you observing the same trend with both `CUDNN=0` & `CUDNN=1` ?

**Comment by lixiangchun** [comment]
I have only tested with `CUDNN=1`.

**Comment by AlexeyAB** [comment]
@lixiangchun 

* What GPU do you use?
* Try to set this value 64: https://github.com/AlexeyAB/darknet/blob/371f21171a2606b4ad5f60651ed6f63cb8d08d99/src/detector.c#L95

**Comment by lixiangchun** [comment]
@AlexeyAB 

* GTX 1080 TI.
* GPU utility is still not as high as `pjreddie/darknet` after set `args.threads=64`.

**Comment by AlexeyAB** [comment]
@lixiangchun You can try to train using flag `-dont_show` in my fork:
`./darknet detector train data/obj.data yolo-obj.cfg darknet19_448.conv.23 -dont_show`

if it will increase GPU-usage, then I will optimize the display of the loss chart.

**Comment by AlexeyAB** [comment]
@lixiangchun I just fixed bug that saves weights for each iteration. It led to a slowdown in training.

**Comment by lixiangchun** [comment]
@AlexeyAB Thanks, it works faster right now.

**Comment by lixiangchun** [comment]
@AlexeyAB Although it works faster than before. I also noticed that `AlexeyAB/darknet` still run a little bit slower than `pjreddie/darknet`.

For example:
**AlexeyAB/darknet**
`Mar 15 11:21 backup/resnet50_yolo_200.weights`
`Mar 15 11:29 resnet50_yolo_800.weights`

Time difference between iteration 200 and 800 is: 8 min.

**pjreddie/darknet**
`Mar 17 17:02 resnet50_yolo_tmp_200.weights`
`Mar 17 17:13 resnet50_yolo_tmp_800.weights`

Time difference iteration 200 and 800 is: 11 min.

**Comment by AlexeyAB** [comment]
@lixiangchun My repo is slower or faster? As I can see time is less when using my fork )

**Comment by lixiangchun** [comment]
@AlexeyAB Sorry, I made a mistake, It should be:

`pjreddie/darknet`
Mar 15 11:21 backup/resnet50_yolo_200.weights
Mar 15 11:29 resnet50_yolo_800.weights

`AlexeyAB/darknet`
Mar 17 17:02 resnet50_yolo_tmp_200.weights
Mar 17 17:13 resnet50_yolo_tmp_800.weights

You repo is slower.

**Comment by AlexeyAB** [comment]
@lixiangchun 

* What `batch, subdivision, width, height` did you use in cfg-file?
* What CUDA, CUDNN version do you use, and what GPU do you use?
* Also what resolution of images do you use for training? If all images of the same size

**Comment by lixiangchun** [comment]
@AlexeyAB 

* `batch = 64, subdivision = 8, width = 416, height = 416, random = 0`
* CUDA v9.1.85 and CUDNN v7.1. (copied from `cudnn-9.1-linux-x64-v7.1.tgz`). GPU is GTX 1080 TI.
* Image size not all the same, but most of them are 720x576.

**Comment by AlexeyAB** [comment]
@lixiangchun I fixed it - added cuDNN for batch normalization as in the original repo. So now training is faster.

**Comment by lixiangchun** [comment]
@AlexeyAB Thanks for your great work.
